### PR TITLE
Feature/update user by id user

### PR DESCRIPTION
**Pull request by isaidou**
# Pull Request Templates

## Type of change

- [x] Feature
- [ ] Fix

### Summary:
This PR introduces the **`update user by ID`** functionality. It allows partial updates to user information while ensuring sensitive fields like passwords are excluded from general updates. 

### Changes:
- Added a new method `updateUserById` in `UserService` to handle partial updates.
- Updated the `UserController` to integrate the new `updateUserById` service method.
- Enhanced the `UserRoutes` to include a **`PUT /api/users/:id`** endpoint for updating user details.

### Motivation:
This change is crucial for allowing the frontend to modify user data efficiently without affecting sensitive fields like passwords. It adheres to API security best practices by restricting which fields can be updated.

### Dependencies:
No new dependencies were added.

## How Has This Been Tested?

- [x] Tested the `PUT /api/users/:id` endpoint using Postman with valid updates.
- [x] Verified proper error handling for invalid user IDs and incomplete payloads.
- [x] Checked the database to ensure only the intended fields are updated.

### Test cases:
1. **Valid Update:**
   - Sent a valid payload to update user details.
   - Verified the success response and updated fields in the database.

2. **Invalid ID:**
   - Sent a non-existent user ID.
   - Verified a `404 Not Found` error response.

3. **Empty Payload:**
   - Sent an empty payload to the endpoint.
   - Verified a validation error response.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/apps/auth/domain/user.service.ts [hunk 1]
```diff
@@ -9,12 +9,34 @@ export class UserService {
...
+  }
+
+  async updateUserById(
+    id: string,
+    updateData: Partial<IUser>
+  ): Promise<UserResponse | null> {
+    const updatedUser = await this.userRepository.updateById(id, updateData);
+    if (!updatedUser) {
+      return null; 
+    }
```
Comment (DaniOmer) [inline src/apps/auth/domain/user.service.ts:45]: tu pourrais throw un `BadRequestError` si !updatedUser

File: src/apps/auth/api/user.controller.ts [hunk 2]
```diff
@@ -20,4 +20,35 @@ export class UserController extends BaseController {
...
+async updateUserById(req: Request, res: Response, next: NextFunction): Promise<void> {
+  try {
+    const { id } = req.params;
+    const updateData = req.body;
+
+    const updatedUser = await this.userService.updateUserById(id, updateData);
+
+    if (!updatedUser) {
+      return ApiResponse.error(res, "User not found", 404, { userId: id });
+    }
```
Comment (DaniOmer) [inline src/apps/auth/api/user.controller.ts:41]: Gérer les erreurs dans le service

**Review by johnnyhelloworld — APPROVED** [review]

File: src/apps/auth/domain/user.service.ts (hunk 1 above)
Comment (isaidou) [reply src/apps/auth/domain/user.service.ts:45]: Prenez moi comme un petit frere

File: src/apps/auth/api/user.controller.ts (hunk 2 above)
Comment (isaidou) [reply src/apps/auth/api/user.controller.ts:41]: fait

**Pull request closed without merging**